modules/samples/extensions/geo-event-fuser/src/main/java/org/wso2/siddhi/extension/geo/geoeventfuser/FuseEvents.java

```diff
@@ -29,66 +26,68 @@
 import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
 import org.wso2.siddhi.core.query.processor.Processor;
 import org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor;
+import org.wso2.siddhi.query.api.definition.Attribute;
 import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.HashMap;
 import java.util.List;
-import org.apache.commons.logging.Log;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class FuseEvents extends WindowProcessor {
```

Comment:
@grainier the class name 'FuseEvents' is a verb or denotes doing something, which is usually not used for a class name. Since we are refactoring this class, shall we fix the name as well? Maybe something like 'FuseEventProcessor' or 'EventFusionProcessor' or similar. wdyt?